src/app/api/common/quorum/getQuorum.ts

```diff
@@ -1,15 +1,14 @@
 import provider from "@/app/lib/provider";
 import prisma from "@/app/lib/prisma";
-import { contracts } from "@/lib/contracts/contracts";
 import { ProposalPayload } from "../proposals/proposal";
-import Tenant from "@/lib/tenant";
+import Tenant from "@/lib/tenant/tenant";
 
 export async function getQuorumForProposal(proposal: ProposalPayload) {
-  const { namespace } = Tenant.getInstance();
+  const { namespace, contracts } = Tenant.getInstance();
```

Comment:
@andreitr,

if these are being read from the process env, then we're not expecting these to change over the process lifetime. we could potentially DRY this by loading the ns and contracts once per process as well. 

this way, we can have the tenant namespace and contracts "resolved" before we enter these queries. looking forward to "dynamic" tenant resolution (i.e. retrieving details at runtime rather than from process env) we would also have a clearer separation of concerns between tenant resolution and query execution, which would allow for easier instrumentation of each of these.

but this can also be an incremental change, just a thought for the future

Reply:
@Flip-Liquid good catch, I am sorta getting around this issue using a singleton, which should initialize the contracts and other configs, only once. The performance hit should be minimal unless nextjs does something funky with back-end vs front-end initialization.